### PR TITLE
Add cheyenne gnu and pgi and other features

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Add cheyenne gnu and pgi machines and ICE_CPPDEFS features
- [X] Developer(s): 
    apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    full test suite passes on cheyenne, 3 compilers, bit-for-bit, https://github.com/CICE-Consortium/Test-Results/wiki/icepack_by_hash_forks#addc97e2e2faee771e9ac3aa9ea1948d8c92fcd4
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on CICE or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please provide any additional information or relevant details below:

- Add cheyenne pgi and gnu
- Clean up pio stuff in Macros files, not needed for icepack
- Remove theia machine files
- Add ICE_CPPDEFS in icepack.settings to support user defined cpps
- Copy env and icepack.settings file to run directory
- Update documentation

### CODE REVIEW

File: doc/source/user_guide/ug_running.rst [hunk 1]
```diff
@@ -78,8 +78,13 @@ Some hints:
...
 - To change namelist, manually edit the **icepack_in** file
 - To change batch settings, manually edit the top of the **icepack.run** or **icepack.test** (if running a test) file
+- When the run scripts are submitted, the current **icepack_in**, **icepack.settings**, and **env.[machine]** files are copied from the case directory into the run directory.  Users should generally not edit files in the run directory as these are overwritten when following the standard workflow.  **icepack.settings** can be sourced to establish the case values in the login shell.  An alias like the following can be established to quickly switch between case and run directories::
+
+    alias  cdrun 'cd `\grep "setenv ICE_RUNDIR"  icepack.settings | awk "{print "\$"NF}"`'
+    alias cdcase 'cd `\grep "setenv ICE_CASEDIR" icepack.settings | awk "{print "\$"NF}"`'
+
 - To turn on the debug compiler flags, set ``ICE_BLDDEBUG`` in **icepack.setttings** to true
-- To change compiler options, manually edit the Macros file
+- To change compiler options, manually edit the Macros file.  To add user define CPP build flags, users can modify ``ICE_CPPDEFS`` in **icepack.settings**.
```
Comment (eclare108213) [inline doc/source/user_guide/ug_running.rst:87]: typo: define should be defined

Comment (phil-blain) [inline doc/source/user_guide/ug_running.rst:87]:
```suggestion
- To change compiler options, manually edit the Macros file.  To add user defined preprocessor macros, users can modify ``ICE_CPPDEFS`` in **icepack.settings**.
```

I'd suggest using standard software engineering terminology, a quick Google search reveals "cppdefs" is not much used outside the climate/weather/environmental modelling communities...

File: doc/source/user_guide/ug_running.rst [hunk 2]
```diff
@@ -693,7 +698,11 @@ Run Directories
 
 The **icepack.setup** script creates a case directory.  However, the model 
 is actually built and run under the ``ICE_OBJDIR`` and ``ICE_RUNDIR`` directories
-as defined in the **icepack.settings** file.
+as defined in the **icepack.settings** file.  It's important to note that when the
+run scripts are submitted, the current **icepack_in**, **icepack.settings**, and **env.[machine]**
```
Comment (eclare108213) [inline doc/source/user_guide/ug_running.rst:702]: "... when the run script is submitted..."  ?

File: doc/source/user_guide/ug_case_settings.rst [hunk 3]
```diff
@@ -39,6 +39,8 @@ to support the CICE model.
    "ICE_DRVOPT", " ", "unused", "icepack"
    "ICE_IOTYPE", " ", "unused", "none"
    "ICE_CLEANBUILD", "true,false", "automatically clean before building", "true"
+   "ICE_CPPDEFS", " ", "user defined cppdefs for build", "null"
```
Comment (phil-blain) [inline doc/source/user_guide/ug_case_settings.rst:42]:
```suggestion
   "ICE_CPPDEFS", " ", "user defined preprocessor macros for build", "null"
```
This seems to be the standard terminology: 
https://gcc.gnu.org/onlinedocs/gcc/Preprocessor-Options.html#Preprocessor-Options

Also, maybe it is worth it to mention that the `-D` flag must be used, i.e. `-DCICE_MACRO` and not just `CICE_MACRO` ?

File: configuration/scripts/machines/env.cheyenne_gnu [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/bin/csh -f
+
+source /glade/u/apps/ch/opt/lmod/7.2.1/lmod/7.2.1/init/csh
```
Comment (phil-blain) [inline configuration/scripts/machines/env.cheyenne_gnu:3]: Just curious, you seem not to use the 
```csh
set inp = "undefined"
if ($#argv == 1) then
  set inp = $1
endif

if ("$inp" != "-nomodules") then
```

bit it the icepack env files, in contrast to what's done in CICE.. 

Not that much of a deal but thought I'd mention.

File: doc/source/user_guide/ug_running.rst (hunk 1 above)
Comment (phil-blain) [reply doc/source/user_guide/ug_running.rst:87]: aslo I think we can drop "users can", the rest of the bullets are all in the imperative mode.

**Comment by apcraig** [comment]
Thanks for the feedback, I have updated the documentation as suggested.

File: configuration/scripts/machines/env.cheyenne_gnu (hunk 4 above)
Comment (apcraig) [reply configuration/scripts/machines/env.cheyenne_gnu:3]: It's a reasonable question.  I have not migrated all the features from CICE scripts to Icepack.  Another one missing is the reuse of the binary during testing.  I think we can add those feature to Icepack scripts as it seems important.  We do a lot more testing and run a lot more tests in CICE than in Icepack at this time, so have prioritized the CICE scripts with performance upgrades.  I try to keep features (like ICE_CPPDEFS) consistent between the two as best as I can.

**Comment by phil-blain** [comment]
@apcraig this point was not addressed:
> Also, maybe it is worth it to mention that the -D flag must be used, i.e. -DCICE_MACRO and not just CICE_MACRO ?

what do you think ?

**Comment by apcraig** [comment]
> @apcraig this point was not addressed:
> 
> > Also, maybe it is worth it to mention that the -D flag must be used, i.e. -DCICE_MACRO and not just CICE_MACRO ?
> 
> what do you think ?

Good idea, I have updated the documentation again.

**Pull request merged**